src/pyEQL/solution.py

```diff
@@ -3228,3 +3230,79 @@ def get_salt_list(self):  # pragma: no cover
         :py:meth:`get_salt_dict`
         """
         return self.get_salt_dict()
+
+    @classmethod
+    def from_preset(cls, preset: str) -> Solution:
+        """Instantiate a solution from a preset composition
+
+        Args:
+            preset (str): String representing the desired solution.
+              Valid entries are 'seawater', 'rainwater', 'wastewater',
+              'urine', 'normal saline' and 'Ringers lactate'.
+
+        Returns:
+            A pyEQL Solution object.
+        """
+        # Path to the YAML and JSON files corresponding to the preset
+        yaml_path = os.path.join("presets", f"{preset}.yaml")
+        json_path = os.path.join("presets", f"{preset}.json")
+
+        # Check if the file exists
+        if os.path.exists(yaml_path):
+            preset_path = yaml_path
+        elif os.path.exists(json_path):
+            preset_path = json_path
+        else:
+            logger.error("Invalid solution entered - %s" % preset)
+            raise FileNotFoundError(f"Files '{yaml_path}' and '{json_path} not found!")
+
+        # Create and return a Solution object
+        return cls().from_file(preset_path)
+
+    def to_file(self, file_name: str, extension: str = "yaml") -> None:
+        """Saving to a .yaml or .json file.
+
+        Args:
+            file_name (str): The name of the file with the saved Solution
+            extension (str, optional): The extension of the save file.
+              Valid entries are 'yaml' and 'json'.
+              Defaults to 'yaml'.
+        """
+        if extension not in ["json", "yaml"]:
+            logger.error("Invalid extension entered - %s" % extension)
+            return
+        file_path = os.path.join("presets", f"{file_name}.{extension}")
+        if extension == "yaml":
+            solution_dict = self.as_dict()
+            dumpfn(solution_dict, file_path)
+        else:
+            dumpfn(self, file_path)
+
+    def from_file(self, file_path: str) -> Solution | None:
+        """Loading from a .yaml or .json file.
+
+        Args:
+            file_path (str): _description_
+        """
+        if not os.path.exists(file_path):
+            logger.error("Invalid path to file entered - %s" % file_path)
+            return None
+        if "yaml" in file_path.lower():
+            true_keys = [
+                "solutes",
+                "volume",
+                "temperature",
+                "pressure",
+                "pH",
+                "pE",
+                "balance_charge",
+                "solvent",
+                "engine",
+                # "database",
+            ]
+            solution_dict = loadfn(file_path)
+            keys_to_delete = [key for key in solution_dict if key not in true_keys]
+            for key in keys_to_delete:
+                solution_dict.pop(key)
+            return Solution(**solution_dict)
+        return loadfn(solution_dict)
```

Comment:
I believe this can be simplified, because `loadfn` can read both `json` and `yaml` files (if you've tried and had a problem loading .yaml, I'd be interested to know)

So in the `yaml` case, instead of `Solution(**solution_dict)`, you can still just `return loadfn(<xxx.yaml>)`